forge/test/mlir/test_ops.py

```diff
@@ -558,3 +559,36 @@ def forward(self, x):
     
     co_out = [co.to("cpu") for co in co_out]
     assert compare_with_golden_pcc(golden=fw_out, calculated=co_out[0], pcc=0.99)
+
+
+@pytest.mark.xfail(reason="ttnn.embedding op fails while reshaping the input_tensor in TILE_LAYOUT")
+def test_adv_index_embedding_decompostion():
+
+    class ForgeAdvIndex(forge.ForgeModule):
+        def __init__(self, name):
+            super().__init__(name)
+
+        def forward(self, input_tensor, indices):
+            return forge.op.AdvIndex("adv_index_op_1", input_tensor, indices)
+
+    model = ForgeAdvIndex("ForgeAdvIndex")
+
+    # Sample Inputs
+    pt_input_tensor = torch.rand((12, 100)).to(torch.float32)
```

Comment:
Can we test out a bit more shape variants?

Reply:
Sure Nikola